### DISCUSSION TITLE
Sustaining AzuraCast, and the "Why Don't You {x}?" FAQ

### DISCUSSION THREAD

**Discussion by BusterNeece** [AzuraCast/AzuraCast#4609] [General]
Hi! I'm Buster Neece (or Silver Eagle in some online circles), and I'm the creator and lead maintainer of AzuraCast.

Over the years of this project's development, I've been very candid and forthcoming about what I think our strong points are, and where I think we struggle as a project, and one of the latter areas is and has always been sustainability.

In this case, "sustainability" means taking the dozens of hours a week, hundreds of hours a month, and thousands of hours a year that it takes to build and maintain a project as complex and interconnected as AzuraCast is, and making it a viable thing for a developer to actually do. That doesn't mean making the typical tech salary of hundreds of thousands a year, but rather a much lower, but still survivable number. For me, where I live, this means about $20,000 a year, or about $1,600 a month. This is _vastly_ less than my compensation as a senior developer would be elsewhere, but after filing for bankruptcy a few years ago, I live very cheaply and that would keep me alive and meet my basic survival needs.

Many of my techie friends want nice cars, nice houses and to live fairly comfortable lives, but I don't aspire to that. My dream is just to be able to survive working on my own terms, on my own schedule, without answering to a profit motive that crushes my spirit and burns me out, which the tech world spent many years doing to me before this project started.

Lately, though, I'm starting to realize that with AzuraCast, this is certainly not possible now, and it may not indeed _ever_ be possible. There are huge barriers to overcome with any of the options, but every time I talk about the struggle of sustaining this project, I get at least a dozen folks chiming in with their own ideas about how to take this project and flip a profit.

Naturally, we've _already thought of_ all of these big ideas. They're well-established in the world of open-source, because there simply aren't that many ways of sustaining FOSS projects, and sometimes the suggestion that we simply haven't _thought of them_ strikes me as a little demeaning of our intellect, even if I know it isn't meant that way.

With that in mind, here's a Frequently Asked Questions, where each FAQ starts with "Why Don't You", because this is how our most frequent questions are presented to us.

## Why don't you...charge for the software itself?

**The short answer is:** because it would fundamentally change what we are as a project, immediately remove a ton of benefits we enjoy today as a free and open-source project, and require that we spin up an entire licensing, encoding, protection and enforcement setup, as well as a customer support branch for paying customers.

Long answer: There is nothing legally stopping us from saying that, starting tomorrow, our software is commercial-only, paid software, and the version that's out today is the last one we'll release for free. There are, however, a LOT of both ideological and infrastructural implications to saying that.

For one, the moment our software stops being classed as free and open-source, suddenly the myriad of software that powers our infrastructure would vanish out from under us. We'd have to start paying out the wazoo for everything from private GitHub repos, CrowdIn translations, JetBrains IDEs, our hosting provided through DigitalOcean, and more. This would create a period where expenses far outstretched profits in the early stages; some folks could handle this financially, but I certainly can't.

What's more, we don't have the business infrastructure required to handle issuing software licenses, encoding our software for prevention of leaks, handling payment processing or providing the support that we would be obligated to provide in those circumstances.

That's not to mention the ethical and ideological conflict of basically giving up on free and open-source software as we know it and shifting to a for-profit model. Not only would that make the project one that wasn't the one I signed up for in the first place, but it would remove its stance as a "great equalizer" in its industry, allowing powerful software to be used by people regardless of income or social status, and allowing folks to openly contribute back to the software in a number of ways. Other projects have made this leap, sure, but in the end they weren't the same project they started as.

## Why don't you...sell add-on software?

**The short answer is:** because our application is built in PHP and frontend web tech, there's almost nothing we could build where the source code wouldn't just leak online in a short order of time.

We could do what our competitors (like Centova Cast) do and lock down, encode and obfuscate the code of our add-ons, but then we run into the issue above: we don't have any infrastructure in place for issuing licenses, managing enforcement, handling the encoding of that code, or providing customer support, as we would be obligated to do if we started selling paid software.

Unlike the item above, this is not impossible for us, however, and it may end up being something we do, if a solution for the problems above is made available to us. We may revisit this option in the future, especially for things that are reseller-specific, like a WHMCS plugin.

## Why don't you...start a Software-as-a-Service offering?

**The short answer is:** This is an entirely separate business than the one our team is experienced in running, and depending on the jurisdiction it's run in, may involve legal issues with regard to licensing. It is not, however, impossible, and may end up happening.

By trade, I'm a web developer, and this project is supported by people who are also web developers (or just otherwise tech-savvy, i.e. translators, community moderators, etc.). I'm not personally in the business of operating a Software-as-a-Service (basically a paid managed hosted radio solution) company, which involves people who are more experienced on the operations (or "ops") side of things.

Starting up a software-as-a-service offering also means putting down an initial outlay of funds to get the hosting infrastructure, sales frontend and provisioning tool up and running. Given my financial position, I can't put down an initial outlay of any funds at all, which prevents me from doing anything where you "gotta spend money to make money."

While many companies have sprung up and offer AzuraCast hosting out there on the web, I wouldn't want to put my name (or this project's official name) on any hosting service that I didn't feel like met an exceedingly high standard of quality or service. I would have to be confident that our hosting offering was one of the absolute best out there, and to spin that up would involve quite a lot of thought, energy, time and money.

Legally, there's another question that comes up: if we're in the business of hosting radio stations, do we have an obligation to make sure those stations are licensed appropriately in the place they're hosted? Some people say yes, some say no. Some say "it depends on where you're hosting." I still don't know the firm answer to this question, and I wouldn't launch into a business personally without knowing that for sure.

A few folks have reached out to me to offer to help get something like this started, and this may end up being a route we pursue at some point in the near future. None of the items above are dealbreakers, just things that add extra complexity and make it much, much more involved than "just" setting something up.

## Why don't you...charge for professional services?

**The short answer:** We're doing that, and nobody's paying.

This is the route we actually chose to go down first, because it made the best use of our existing experience, it was doable without a huge initial outlay of funds, and the amount of time we spent on it was directly proportional to the amount of sales we actually got, meaning we weren't likely to waste a ton of time and energy on it for no return.

We've started promoting our human services sector across our web presence, not just on our homepage but also in our documentation, alongside the installation instructions. It is currently impossible to make it through to the AzuraCast installation without at least _seeing_ that we offer services (like installation, updates, migrations, etc.) on a paid basis.

The problem is that our analytics are showing remarkably, astonishingly low interest in these offerings.

It's hard to speculate on why this may be the case without speculating a bit on the mindset of our users. The way I see it right now, we're in a paradoxical situation: because AzuraCast is positioned as the "free" (not just as in libre, but gratis, or free-as-in-beer) option, it doesn't tend to attract people who are willing to spend money on getting their radio station project started. At the same time, if we start to target users with paid services, we suddenly are competing up against companies with a lot more money, a lot bigger teams, and a lot more time than us, where we don't hold up as well.

---

## So...what _do_ you do then?

What is the answer for all this?

The thing I'm starting to realize, and that nobody is willing to actually say, is that it's possible the answer is that this piece of software, in this particular industry, just _can't_ sustain itself as an open-source offering.

The operating margins for web radio stations are exceedingly thin, with many stations just barely breaking even despite on-site and in-stream advertisement, on account of costs piling up for hosting, bandwidth and licensing. Basically, for many stations, if there's a cost that _can_ be cut, it _must_ be cut in order to keep the station alive.

Because AzuraCast is seen as a "free" component, anything we could bolt onto the software that added expense to it would, likewise, become another "cuttable" cost for station operators, another elective expense that they would opt out of.

It's possible there's just not enough money in web radio to sustain us. It's not an easy answer, but that might be where we're at.

The thought of that probably shouldn't keep any of you up at night, but it most definitely does to me. For years, I sunk a lot of my hopes and dreams into the project, hoping that it would enable me to rest my weary bones for a while, to work on a project I both cared about, believed in, and enjoyed working on as a full-time job, which I've been doing anyway for years now.

What if that's just not possible? What if I'll always need some "side hustle" to keep the lights on? Well...that's fine, I suppose, but it means we have to be more realistic about the role AzuraCast should play in my life, and how much time and energy I should be sinking into it, and whether that should be much, much less than it is today.

**Comment by gusaus** [comment]
Very informative post - thanks. 

I'm not sure if the suggestions in this older discussion are relevant. https://github.com/AzuraCast/AzuraCast/discussions/3994#discussioncomment-662412

Just wondering if you thought more stations would make recurring contributions if their logo was displayed on the AzuraCast website or even the readme? https://docs.opencollective.com/help/contributing/development/readme-integration

**Comment by countrymusicfy** [comment]
I think one good place to start is to create a collective on Open Collective, and allow every azuracast station to create a project under your collective. It showcases your code and how it is implemented in various projects around the world. It also allows stations to participate in fund raising.

**Reply by gusaus** [reply]
AzuraCast already has https://opencollective.com/azuracast but I'm not sure creating projects for stations makes sense. Stations themselves could however create their own collective. 

I do think creating a page on AzuraCast where sponsoring stations could have their logo displayed would incentivize stations to contribute. Again I left some suggestions in this previous discussion https://github.com/AzuraCast/AzuraCast/discussions/3994#discussioncomment-662412

Would be good to hear from other users/stations regarding what might entice them to support AzuraCast, Buster, and the core team.

**Reply by countrymusicfy** [reply]
I disagree about stations creating their own collectives. That would be a good strategy if you want to isolate and fragment each station from each other. I think the case for an Azuracast Collective is stronger when you see all the projects that make Azuracast a collective in the first place.

**Reply by gusaus** [reply]
I agree grouping together and showcasing AzuraCast stations is a good idea. One way to do that would be to showcase those who contribute financially. https://obsproject.com/contribute provides a good example. 

Aside from the administrative nightmare, the entity and projects feature for https://opencollective.com/azuracast isn't set up to sustain users or stations.

**Reply by countrymusicfy** [reply]
I guess the bigger problem is that the azuracast community is not alarmed by this post, since we are the only ones commenting on it.

**Reply by gusaus** [reply]
I'm not sure how many users are aware the project might not be continuously updated they don't help sustain. I think more might contribute to one of the following if they knew what a huge difference it would make to the life of the project and Buster. 

I don't know if there's been an uptick in services, but providing a prominent space on https://www.azuracast.com/ at least makes it more obvious. Still wondering if doing something similar to https://obsproject.com/contribute would also help.

I'd like to help set that up if there's a need.

**Reply by gusaus** [reply]
Resurfacing this part of the sustainer discussion as AzuraCast is testing the waters on using Github and Open Collective for feature bounties with https://github.com/AzuraCast/AzuraCast/issues/5242

After observing and participating in subsequent discussions on the topic in [AzuraCast chat](https://discord.com/channels/647143522054701075/647143522054701084), I'd like to recommend we look at how open source projects [OBS](https://github.com/obsproject) and [Mautic](https://github.com/mautic) are using tools like Github and Open Collective to encourage and facilitate community contributions. 

- A [documentation section](https://docs.azuracast.com/en/home) using [Mautic's RFP process](https://contribute.mautic.org/product-team/request-for-proposals-rfp-process) or [OBS Project Bounty Program](https://github.com/obsproject/obs-studio/wiki/OBS-Project-Bounty-Program) as a model. 
- [Github discussion and category](https://github.com/AzuraCast/AzuraCast/discussions) similar to [Mautic](https://github.com/mautic/mautic/discussions/categories/requests-for-proposals-rfps) and [OBS](https://github.com/obsproject/obs-studio/discussions/categories/requests-for-proposals-bounties) for RFPs (Bounties)
- [Open Collective projects](https://opencollective.com/azuracast/projects) similar to [Mautic](https://opencollective.com/mautic/projects) or [OBS](https://opencollective.com/obsproject/projects/obs-project-bounty-program) to allocate and/or raise funds for projects where funds are needed to move forward.

Curious to hear thoughts on those recommendations from Buster and the team. If there's something worth incorporating, I'd like to bring in help from [OpenProducer](https://github.com/OpenProducer/community) and other folks/partners in the [Open Collective community](https://join.slack.com/t/opencollective/shared_invite/zt-f43qko76-sD8G~e_vQCm4TtpIsM4i~A) move it forward.

**Comment by Moonbase59** [comment]
These are questions not easily answered, as Buster shows us above. I advocate FOSS and strongly believe that knowledge should be shared freely, but also that people should _share if they can_. After all, we do get something out of it (and also probably invest a lot of time finding other’s bugs). But as long as mankind evolves ever more into "el cheapo" and "grab-n-run" mentality, it will probably never happen that shares are compensated as in "I get a great piece of software" and, when using it for whatever purpose and gaining something from it "I care and give something back, now let’s see that we can sustain the developer. He needs food, after all." Instead, the greedy win and sack all profits.

I myself once had high-paying jobs, and none today. Struggling to live, partly because I also tried that route and believed that something would come back. Yeah, it did. More work. No pay. Time I should probably have better invested in a new job, or being happy instead of working day and night for nothing. It ain’t easy. Really.

Well, enough lament. Let’s do what we can. Together, not against each other. As long as we can.

(And no, I’m not a communist or something. Just sometimes the idea creeps up how wonderful and happy our little globe and its inhabitants could be, if we _cared_.)

**Reply by gusaus** [reply]
Totally forgot about this discussion! Good timing as I had just updated a [related discussion](https://github.com/orgs/OpenProducer/discussions/25#discussioncomment-5184225) about sustaining AzuraCast and other open source projects in the Open Collective ecosystem.

A quick skim through some of the [open source projects](https://discover.opencollective.com/?tag=open+source) on Open Collective should make it obvious that most small, independent open source projects have difficulty sustaining. 

That said, I've always believed (and still do) that such a solid, popular tool like AzuraCast would be too valuable for the huge community of users who rely on it to 'not' help Buster and the team keep it around. 

Maybe there still needs to be some additional functionality (seems like you covered some of those in https://github.com/AzuraCast/AzuraCast/issues/4460#issuecomment-1435697282) that would make it a viable solution for users with bigger budgets (like radio stations). If we can circle back to my last comment https://github.com/AzuraCast/AzuraCast/discussions/4609#discussioncomment-2540721, maybe we can band together and create a couple of feature campaigns.

**Reply by Moonbase59** [reply]
I wasn’t aware of [Open Collective](https://opencollective.com/) and informed myself a little. This might indeed be a viable solution/way to go.

**Comment by starapple2** [comment]
I'm a retired journalist/producer hastened out of the mainstream media industry by emerging Web technologies. I host a news features site using WordPress served from a local machine.

Like Buster, mine is a labour of love using available resources. For many years I have toyed with Icecast and Shoutcast, mainly playing the reggae music I like and telling only close friends when streams are available as the cost to establish a registered "station" doesn't tally with the absence of revenue.

Recently I thought I'd attempt talk radio as the trend in music listening these days is curated playlists abd other on-demand methods, while talk is dominated by on-demand podcasts. Only while driving am I listening to live radio.

All traditional media are struggling to make money and so are using their resources to get into the on-demand space, with terrestrial radio directing their output to online streams. That latter audience is likely composed mostly of expat communities who want to tune in to happenings at home.

But my love for and training in mass communication specializing in broadcast radio means I am hooked to the idea of radio however delivered.

I have been toying with contacting former colleagues to join a collective linked by a Web interface so all do not need to download and install streaming servers and software. I started writing HTML5/JavaScript media interfaces and bumped into the need for sockets and NodeJS and similar tech that eventually led me to Liquidsoap and LibreTime and then AzuraCast.

AzuraCast seemed the most user friendly of the lot of open-source offerings and I finally succeeded in installing it unlike the incompatibluties encountered with LibreTime.

I have summarized all this to say:
1. Radio is not a growing media segment whether terrestrial, satellite or streaming
2. The technology to enter the streaming segmentin terms of the server and presenter (DJ) are easily accessible.
3. A difficulty arises to make the service "distributed" among collaborators.
4. Documentation for most of the elements beyond Icecast/Shoutcast is poor and not usually friendly to end-users and that is often cryptic and still assuming a high level of tech competence. For example, nothing is presented in a procedural way and is often lost in the developers' expose on pertinent but not relevant info to do things.
5. The devs need to be more aware of who the end users are likely to be.
6. Improved manuals increase the value of the software and offer a chance of an income stream in the way I imagine Liquidsoap is attempting with its Cookbook.

I did a technical writing and editing course here in Toronto a few years ago and would be more than willing to be a collaborator in the area of instructions manuals for better enduser experience and to provide those involved in development with an income stream.

Best to all.

**Reply by gusaus** [reply]
I shared your comment in Discord and mentioned it doesn't seem like there's a clear/obvious entry point for contributors on https://github.com/AzuraCast or https://www.azuracast.com/

This was the feedback from one the maintainers.

> If they're relating to the docs, they can simply fork our documentation and make the appropriate proposals for changes. Documentation is a tricky subject to handle because there's no such thing as 'perfect documentation', we try our best with what availability we have.

Maybe contributing to improving the onboarding experience could be an area you could help with?

If you skim through some of the previous comments, I'm mentioning a couple other collaboration and contribution areas in the Open Collective community.
